scripts/lint.sh

```diff
@@ -60,12 +60,25 @@ lint(){
         printf "%s\n" "Require golangci-lint. You can run this command in a docker container instead with '-c' and not worry about it or install it: https://golangci-lint.run/usage/install/"
     fi
 
+    # Group the lint warnings together. If errexit is enabled, we will only see
+    # the results of the sdk lints if the node lints have no violations.
+    # It's more helpful to see all of the violations at once.
+    set +e
+
     # Do the actual linting!
-    cd "$ROOT"/node
+    cd "${ROOT}/node"
     golangci-lint run --timeout=10m $GOLANGCI_LINT_ARGS ./...
+    node_lint_status=$?
 
     cd "${ROOT}/sdk"
     golangci-lint run --timeout=10m $GOLANGCI_LINT_ARGS ./...
+    sdk_lint_status=$?
+
+    # Restore errexit
+    set -e
+
+    # Return failure if either linting operation failed
+    [[ $node_lint_status -eq 0 && $sdk_lint_status -eq 0 ]]
```

Comment:
Should this be something like:

```
[[ $node_lint_status -eq 0 && $sdk_lint_status -eq 0 ]] || exit 1
```

instead?

Reply:
Yeah I think so. I don't think `[[ ]]`  will trigger an exit even with `set -e`

Reply:
```suggestion
    [[ $node_lint_status -eq 0 && $sdk_lint_status -eq 0 ]] || exit 1
```
I'll make the change. Right now the script when run with the "lint" command will return 1 when the linters fail, as expected. That happens I think because the return value of `lint()` is returned to the caller (the `case` statement at the end of the script), which is the last statement in the code. Exiting early here more closely matches what `set -e` is probably intending to do. Without the exit, the script could theoretically continue running after a failing lint if it was modified to have other statements after the `case` part.